### PR TITLE
metrics: for job submit events

### PR DESCRIPTION
**Pull request by frrist**

### CODE REVIEW

File: pkg/analytics/models.go [hunk 1]
```diff
@@ -138,3 +139,88 @@ func NewJobTerminalEvent(j models.Job) *Event {
...
+		TaskGPUCount:         taskResources.GPU,
+		TaskNetworkType:      t.Network.Type.String(),
+		TaskDomainsCount:     len(t.Network.Domains),
+		TaskExecutionTimeout: t.Timeouts.ExecutionTimeout,
+		TaskQueueTimeout:     t.Timeouts.QueueTimeout,
+		TaskTotalTimeout:     t.Timeouts.TotalTimeout,
+		Warnings:             warnings,
+	}
+
+	// return NewEvent(SubmitJobEventType, submitEvent)
```
Comment (wdbaruni) [inline pkg/analytics/models.go:225]: commented code

File: pkg/analytics/models.go [hunk 2]
```diff
@@ -138,3 +139,88 @@ func NewJobTerminalEvent(j models.Job) *Event {
...
+		TaskCPUUnits:         taskResources.CPU,
+		TaskMemoryBytes:      taskResources.Memory,
+		TaskDiskBytes:        taskResources.Disk,
+		TaskGPUCount:         taskResources.GPU,
+		TaskNetworkType:      t.Network.Type.String(),
+		TaskDomainsCount:     len(t.Network.Domains),
+		TaskExecutionTimeout: t.Timeouts.ExecutionTimeout,
+		TaskQueueTimeout:     t.Timeouts.QueueTimeout,
+		TaskTotalTimeout:     t.Timeouts.TotalTimeout,
+		Warnings:             warnings,
```
Comment (wdbaruni) [inline pkg/analytics/models.go:222]: can this contain PII

File: pkg/orchestrator/endpoint.go [hunk 3]
```diff
@@ -52,14 +53,20 @@ func NewBaseEndpoint(params *BaseEndpointParams) *BaseEndpoint {
...
+	submitEvent := analytics.NewSubmitJobEvent(*job, warnings...)
+	defer func() {
+		analytics.EmitEvent(ctx, analytics.NewEvent(analytics.SubmitJobEventType, submitEvent))
+	}()
 
 	if err := e.jobTransformer.Transform(ctx, job); err != nil {
+		submitEvent.Error = err.Error()
 		return nil, err
 	}
+	submitEvent.ID = job.ID
```
Comment (wdbaruni) [inline pkg/orchestrator/endpoint.go:69]: nice

File: pkg/analytics/models.go [hunk 4]
```diff
@@ -14,7 +14,8 @@ import (
 type EventType string
 
 const (
-	JobEventType = "job_v1"
+	JobEventType       = "job_v1"
+	SubmitJobEventType = "submit_job_v1"
```
Comment (wdbaruni) [inline pkg/analytics/models.go:18]: based on feedback from Peter, lets name these `bacalhau.job_v1.start` and `bacalhau.job_v2.stop` or something on those lines

**Review by wdbaruni — APPROVED** [review]
minor comments. you can merge after addressing them

File: pkg/analytics/models.go (hunk 2 above)
Comment (frrist) [reply pkg/analytics/models.go:222]: No, not currently.

**Pull request merged**